mockgcp/Makefile

```diff
@@ -12,16 +12,15 @@ tools:
 	wget -N -O bin/protoc.zip https://github.com/protocolbuffers/protobuf/releases/download/v3.12.4/protoc-3.12.4-linux-x86_64.zip
 	cd bin; rm -rf protoc; mkdir protoc; cd protoc; unzip ../protoc.zip
 
-GOOGLEAPI_VERSION?=HEAD
 .PHONY: sync-repo
```

Comment:
Any reasons to remove this feature?  Dedicating to a specific one is better for version control

Reply:
In my understanding, `make sync-repo` should be a target to sync the repo to remote master HEAD. But what `cd third_party/googleapis && git fetch --all && git reset --hard HEAD` does is not syncing the repo to match remote, instead, it simply checkout the local HEAD without rebasing the remote branch. So, not syncing at all.

Or, I can create a new target `checkout-head` to represent the old behavior of `sync-repo`.

Reply:
Hmmm, so here is a reason I made the change:

I don't know what's the original purpose of `make sync-repo` is, but the functionality to sync local repo `./mockgcp/third_party/googleapis` to remote master HEAD is missing, and can be addressed by `git checkout -f origin/master` (creating a detached branch, but not actually updating the local branch) or `git reset --hard origin/master` (updating the local branch to match the remote master HEAD). (I picked the latter because I think we should update the local branch.)

In mockgcp doc (https://github.com/GoogleCloudPlatform/k8s-config-connector/tree/master/mockgcp#adding-a-new-service), it said: `Add the proto to the Makefile and run make gen-proto (or just make).` While it didn't say what contributors should expect after running the command, I expected that the proto files for the exact resource I worked on under `./mockgcp/generated/mockgcp` would be updated and match the remote master HEAD.

However `make gen-proto` doesn't update `./mockgcp/generated/mockgcp` at all.

My change is to fix this.

Reply:
Rebased on top of master and reverted such change. @justinsb seems to be improving the process right now.